sormas-ui/src/main/java/de/symeda/sormas/ui/caze/CaseDataForm.java

```diff
@@ -864,32 +870,108 @@ protected void addFields() {
+				}
+			} else if (followUpDate != null) {
+				if (newQuarantineEnd.compareTo(followUpDate) > 0) {
+					confirmQuarantineEndExtended(quarantineExtendedCheckBox, quarantineReducedCheckBox, quarantineEndField, originalCase, followUpDate, followUpUntilField);
+				} else if (newQuarantineEnd.compareTo(followUpDate) < 0) {
+					confirmQuarantineEndReduced(quarantineExtendedCheckBox, quarantineReducedCheckBox, quarantineEndField, originalCase, followUpDate);
+				}
+			}
+		} else if (!originalCase.isQuarantineExtended() && !originalCase.isQuarantineReduced()) {
+			setVisible(false, quarantineExtendedCheckBox.getId(), quarantineReducedCheckBox.getId());
+			quarantineExtendedCheckBox.setValue(false);
+			quarantineReducedCheckBox.setValue(false);
+		}
+	}
+
+	private void confirmQuarantineEndExtended(
+		CheckBox quarantineExtendedCheckbox,
+		CheckBox quarantineReducedCheckbox,
+		Property<Date> quarantineEndField,
+		CaseDataDto originalCase,
+		Date oldQuarantineEnd,
+		DateField followUpUntil) {
+		VaadinUiUtil.showConfirmationPopup(
+			I18nProperties.getString(Strings.headingExtendQuarantine),
+			new Label(I18nProperties.getString(Strings.confirmationExtendQuarantine)),
+			I18nProperties.getString(Strings.yes),
+			I18nProperties.getString(Strings.no),
+			640,
+			confirmed -> {
+				if (confirmed) {
+					originalCase.setQuarantineExtended(true);
+					originalCase.setQuarantineReduced(false);
+					quarantineExtendedCheckbox.setValue(true);
+					quarantineReducedCheckbox.setValue(false);
+					setVisible(true, ContactDto.QUARANTINE_EXTENDED);
+					setVisible(false, ContactDto.QUARANTINE_REDUCED);
+					if (originalCase.getFollowUpUntil() != null) {
+						confirmExtendFollowUpPeriod(originalCase, quarantineEndField.getValue(), followUpUntil);
+					}
+				} else {
+					quarantineEndField.setValue(oldQuarantineEnd);
+				}
+			});
+	}
+
+	private void confirmExtendFollowUpPeriod(CaseDataDto originalCase, Date quarantineEnd, DateField followUpUntil) {
+		if (quarantineEnd.compareTo(originalCase.getFollowUpUntil()) > 0) {
 			VaadinUiUtil.showConfirmationPopup(
-				I18nProperties.getString(Strings.headingExtendQuarantine),
-				new Label(I18nProperties.getString(Strings.confirmationExtendQuarantine)),
+				I18nProperties.getString(Strings.headingExtendFollowUp),
+				new Label(I18nProperties.getString(Strings.confirmationExtendFollowUp)),
 				I18nProperties.getString(Strings.yes),
 				I18nProperties.getString(Strings.no),
 				640,
 				confirmed -> {
 					if (confirmed) {
-						if (!originalCase.isQuarantineExtended()) {
-							quarantineExtendedCheckbox.setValue(true);
-						}
-					} else {
-						quarantineEndField.setValue(oldQuarantineEnd);
+						originalCase.setFollowUpUntil(quarantineEnd);
+						followUpUntil.setReadOnly(false);
+						followUpUntil.setValue(quarantineEnd);
+						followUpUntil.setReadOnly(true);
```

Comment:
followUpUntil is not always readonly, here you force it to readonly.

See the checkbox  "Overwrite follow-up until date" besides the date field